src/mlir/CMakeLists.txt

```diff
@@ -1,19 +1,22 @@
 find_package(Threads REQUIRED)
 
 # We're not building LLVM, but downloading the build dir from an artefact
-set(LLVM_BUILD "${CMAKE_SOURCE_DIR}/external/llvm-project/build")
+set(LLVM_BUILD ${VERONA_LLVM_LOCATION})
+
+# std::iterator is used by MLIR/LLVM
```

Comment:
I'd expand on this a bit, saying a fix in LLVM is probably going to happen eventually and we need to revisit this warning when it does.